libs/datasets/timeseries.py

```diff
@@ -297,14 +311,17 @@ class MultiRegionTimeseriesDataset:
     @staticmethod
     def from_csv(path_or_buf: Union[pathlib.Path, TextIO]) -> "MultiRegionTimeseriesDataset":
         df = common_df.read_csv(path_or_buf, set_index=False)
+        _add_location_id(df)
```

Comment:
Would it be possible to add this in at a single point? maybe in `from_source_data` or when we're constructing combined datasest?  I would love to be in a position where it was already included in the saved csv instead of having to mutate the loaded csv

Reply:
I made the change to put location_id in the combined datasets. I really like the idea that the saved csv and in-memory representation are separate, mutating when loading is very inexpensive and as long as it is a 1-1 transformation between representations doesn't reduce our observalibity for debugging much. I was motivated by a desire to move location_id further upstream.